src/main/java/com/vestmark/bitbucket/plugin/MergeConflictDetectorService.java

```diff
@@ -0,0 +1,205 @@
+    private static final String AUTO_MERGE_FAIL = "Automatic merge failure";
+    private static String HOST_URL;
+
+    private final ApplicationPropertiesService applicationPropertiesService; 
+    private final AuthenticationContext authenticationContext;
+    private final GitExtendedCommandFactory extendedCmdFactory;
+    private final PullRequestService pullRequestService;
+    private final RefService refService;
+    private final BranchModelService modelService;
+    private final VersionComparator<Branch> branchComparator;
+
+    @Autowired
+    public MergeConflictDetectorService(@ComponentImport ApplicationPropertiesService applicationPropertiesService,
+                                        @ComponentImport PullRequestService pullRequestService, 
+                                        @ComponentImport GitExtendedCommandFactory extendedCmdFactory, 
+                                        @ComponentImport AuthenticationContext authenticationContext, 
+                                        @ComponentImport RefService refService,
+                                        @ComponentImport BranchModelService modelService) 
+   {
+      this.applicationPropertiesService = applicationPropertiesService;
+      this.authenticationContext = authenticationContext;
+      this.extendedCmdFactory = extendedCmdFactory;
+      this.pullRequestService = pullRequestService;
+      this.refService = refService;
+      this.modelService = modelService;
+      branchComparator = new VersionComparator<Branch>(Branch::getDisplayId);
+    }
+
+    @PostConstruct
+    private void postConstruct() {
+      HOST_URL = applicationPropertiesService.getBaseUrl().toString();
+    }
+
+    @GET
+    @AnonymousAllowed
+    @Produces({MediaType.APPLICATION_JSON})
+    @Path("/{repoIdStr}/{pullRequestIdStr}")
+    public Response getMergeResults(@PathParam("repoIdStr") String repoIdParam, @PathParam("pullRequestIdStr") String pullRequestIdParam)
+    {
+      int repoId = Integer.parseInt(repoIdParam);
+      long pullRequestId = Long.parseLong(pullRequestIdParam);
+      MergeConflictDetector mcd = new MergeConflictDetector(authenticationContext.getCurrentUser(), 
+        pullRequestService.getById(repoId, pullRequestId), HOST_URL);
+      BranchClassifier bc = modelService.getModel(mcd.getToRepo()).getClassifier();
+      BranchType toBranchType = bc.getType(mcd.getToBranch());
+      // If the target is not master and is a release branch, find target branch and upstream releases (if any).
+      if (!mcd.getToBranchId().equals("refs/heads/master") && toBranchType != null && toBranchType.getId().equals("RELEASE") ) {
+        Page<Branch> branches = bc.getBranchesByType(toBranchType, new PageRequestImpl(0,PageRequestImpl.MAX_PAGE_LIMIT/2));
+        branches.stream()
+              .parallel()
+              .filter(b -> mcd.isRelated(b))
+              .filter(b -> mcd.isUpstreamBranch(b))
+              .sorted(branchComparator)
```

Comment:
I just tried it both ways and there is no difference in the final results.

**However**, if I remove this line, the overall mcd duration on a PR (targeted at 8.2 for instance) takes ~18 seconds instead of ~15.  So it is faster sorting twice and I can't figure out why.  There are 6 release branches between 8.2 and master.  What I am seeing in the logs is that the parallelization gets throttled after the first 3 dryRuns get kicked off with this line removed.  Hence the extra 3 seconds.  It doesn't make any logical sense to me though.

With ".sorted(branchComparator)" all DryRuns for release branches initiate at the same time:
Action start Fri Aug 07 13:29:37 EDT 2020
DryRun start Fri Aug 07 13:29:37 EDT 2020, Comparing release/8.3.5
DryRun start Fri Aug 07 13:29:37 EDT 2020, Comparing release/8.3
DryRun start Fri Aug 07 13:29:37 EDT 2020, Comparing release/8.2
DryRun start Fri Aug 07 13:29:37 EDT 2020, Comparing release/8.2.5
DryRun start Fri Aug 07 13:29:37 EDT 2020, Comparing release/8.5
DryRun start Fri Aug 07 13:29:37 EDT 2020, Comparing release/8.5.5
DryRun start Fri Aug 07 13:29:41 EDT 2020, Comparing master
Action complete Fri Aug 07 13:29:52 EDT 2020


Without ".sorted(branchComparator)" last 3 DryRuns for release branches are delayed for some reason:
Action start Fri Aug 07 13:32:39 EDT 2020
DryRun start Fri Aug 07 13:32:39 EDT 2020, Comparing release/8.3
DryRun start Fri Aug 07 13:32:39 EDT 2020, Comparing release/8.5
DryRun start Fri Aug 07 13:32:39 EDT 2020, Comparing release/8.2
DryRun start Fri Aug 07 13:32:40 EDT 2020, Comparing release/8.2.5
DryRun start Fri Aug 07 13:32:41 EDT 2020, Comparing release/8.3.5
DryRun start Fri Aug 07 13:32:43 EDT 2020, Comparing release/8.5.5
DryRun start Fri Aug 07 13:32:46 EDT 2020, Comparing master
Action complete Fri Aug 07 13:32:57 EDT 2020